staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator/aggregator.go

```diff
@@ -118,6 +100,31 @@ func BuildAndRegisterAggregator(downloader *Downloader, delegationTarget server.
 	return s, nil
 }
 
+type specAggregator struct {
+	// mutex protects all members of this struct.
+	rwMutex sync.RWMutex
+
+	// Map of API Services' OpenAPI specs by their name
+	openAPISpecs map[string]*openAPISpecInfo
+
+	// provided for dynamic OpenAPI spec
+	openAPIVersionedService *handler.OpenAPIService
+}
+
+var _ SpecAggregator = &specAggregator{}
+
+// This function is not thread safe as it only being called on startup.
+func (s *specAggregator) addLocalSpec(spec *spec.Swagger, localHandler http.Handler, name, etag string) {
```

Comment:
can we add `s. addLocalSpecCalled = true` and then panic when this function is called again?

Reply:
or rename this to `addLocalSpecOnce` ?

Reply:
Why?

It's only called from the speAggregator constructor `BuildAndRegisterAggregator`. What are you fearing here?

Reply:
@sttts fearing every function that has "should be only called once on startup"  :-)

Reply:
I will point out that the function is private and restricted to be used in this file. For that I won't add a once to be called logic. Also it may run more than once for local specs on startup.